src/x86.rs

```diff
@@ -184,21 +188,35 @@ pub struct Avx2Searcher<N: Needle> {
 }
 
 impl<N: Needle> Avx2Searcher<N> {
-    /// Creates a new searcher for `needle`. By default, `position` is
-    /// set to the last character in the needle.
+    /// Creates a new searcher for `needle`. By default, `position` is set to
+    /// the last character in the needle.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `needle` is empty or if the associated `SIZE` constant does
+    /// not correspond to the actual size of `needle`.
     #[target_feature(enable = "avx2")]
     pub unsafe fn new(needle: N) -> Self {
-        let position = needle.size() - 1;
+        // Wrapping prevents panicking on unsigned integer underflow when
+        // `needle` is empty.
+        let position = needle.size().wrapping_sub(1);
```

Comment:
Would `saturating_sub` makes more sense here? Not that it really matters since the empty needle case is going to panic anyway.

Reply:
I went with `wrapping_sub` as I imagined it will be more performant on most platforms although I think you're right that `saturating_sub` is more intuitive. I'm happy with either, what do you think?

Reply:
I think you are right that it should be more performant and since it does not really matter, let's keep it that way for now.